File: python_solution/53.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # by kadane algo O(n)
        maxlocal = maxglobal = float('-inf')
        for i in nums:
            maxlocal = max(maxlocal+i,i)
            maxglobal = max(maxglobal,maxlocal)
        return maxglobal





#[-2,1,-3,4,-1,2,1,-5,4]

File: C_solution/53.cpp
#include<vector>
#include <algorithm>
#include<limits>
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int localmax = 0, globalmax = INT_MIN;
        for (int n:nums){
            localmax = max(localmax+n,n);
            globalmax = max(localmax,globalmax);
        }
        return globalmax;
    }
};